### ISSUE TITLE
Should the selected type for stringInt column be string instead of string | number ? 

### ISSUE THREAD

**Issue by lorefnon** [juanluispaz/ts-sql-query#108]
Hello, hope you are doing well. 

For a sample table like this: 

```ts
export class UserTable extends Table<DBConnection, 'UserTable'> {
    id = this.autogeneratedPrimaryKey('id', 'stringInt');
}

export type UserIRow = InsertableRow<UserTable>;
export type UserURow = UpdatableRow<UserTable>;
export type UserSRow = SelectedRow<UserTable>;
```

It is understandable that `UserIRow["id"]` and `UserURow["id"]` is `number | string` , but should `UserSRow["id"]` be `number | string` ? Will the selected value not be always a string ?

**Comment by juanluispaz** [comment]
Hi,

I'm working on allowing you to have your own types to represent any specialised value, like `int` or `double`; right now, you can define `customComparable`, but you will lose the `add` or `subtract` functionality. I want you to have full control over the projected value losing the minimal functionality. I'm working on adding `customInt`, `customDouble`, ... columns.

Meanwhile, you can use `customComparable` columns in combination with SQL fragments to deal with it; or you can create a TS function that post-process the result.

I believe that `customInt` or `customDouble` will allow you to do what you need.

Let me know if that looks good for you.

**Comment by lorefnon** [comment]
> you can create a TS function that post-process the result

Yes, this is what I am currently doing and it's not too much of a hassle. I just felt like the string | number union in selected type could have been something unintended that got missed, so wanted to report.

The upcoming customInt support looks interesting though.

**Comment by juanluispaz** [comment]
The reason for `string | number` union comes from most of the databases connectors trying to return a number when the column represents a number but fallback to a string when the number cannot be represented in a feasible way in javascript (this depends on the connector you uses); in consequence, you end with a `string | number` representation in the projected value.

With `customInt` and `customDouble` will open the door to represent the value always as a string, or even better, using an opaque type.

**Comment by lorefnon** [comment]
Ah I see, understandable in that case.  Looking forward to the inclusion of customInt. Thanks a lot.

**Comment by juanluispaz** [comment]
Released ts-sql-query 1.59.0 adding support to more custom types: `customInt`, `customDouble`, `customUuid`, `customLocalDate`, `customLocalTime`, `customLocalDateTime`.

**Comment by juanluispaz** [comment]
Hi,

 ts-sql-query 1.59.0 added support to more custom types. I'm closing this issue as completed.
